frontend_gen.py

```diff
@@ -222,7 +283,11 @@
         app.write(app_import_template.format(page_name=page.replace(" ", "_")))
     app.write(
         app_export
-        + app_render_header.format(
+        + (
+            app_render_header
+            if values_yaml["SciViz"]["auth"]
+            else app_render_header_nologin
```

Comment:
Similarly to how we did in Pharus, we can probably push this a bit higher since we perform the check repeatedly.

Reply:
This code is not in a for loop (meaning it is only traversed once). The only file that is different if there is no login is the app.ts, since this file is only created once the check for auth is only traversed once.

Reply:
Never mind, there is another spot where we can optimize it like we did in pharus. I will make the change.